broadinstitute/scp-rails-baseimage
# Upgrading to Ruby 3.2, Node 20 (SCP-5478)

#### BACKGROUND & CHANGES
This updates the base image to using Ruby 3.2.2 and Node 20, as well as updating to v2.5.1 of `phusion/passenger-docker`.  Importantly, this also changes the build type from `full` to `ruby32`.  This skips installing previous versions of Ruby, as well as Python and other frameworks that are not used.  We previously relied on the `build_full` to provision Node, but as we manually install this later to control the version better, we no longer need the full build.  This reduces the size of the resulting image by ~100MB.  Difference in builds can be seen here (from the phusion-passernger [Makefile](https://github.com/phusion/passenger-docker/blob/rel-2.5.1/Makefile)):

```
build_ruby32: build_base
	rm -rf ruby32_image
	cp -pR image ruby32_image
	echo ruby32=1 >> ruby32_image/buildconfig
	echo final=1 >> ruby32_image/buildconfig
...

build_full: build_base
	rm -rf full_image
	cp -pR image full_image
	echo ruby27=1 >> full_image/buildconfig
	echo ruby30=1 >> full_image/buildconfig
	echo ruby31=1 >> full_image/buildconfig
	echo ruby32=1 >> full_image/buildconfig
	echo jruby93=1 >> full_image/buildconfig
	echo jruby94=1 >> full_image/buildconfig
	echo python=1 >> full_image/buildconfig
	echo nodejs=1 >> full_image/buildconfig
	echo redis=1 >> full_image/buildconfig
	echo memcached=1 >> full_image/buildconfig
	echo final=1 >> full_image/buildconfig
```

In addition, this updates to newer Github Action version to get off of deprecated Node runtimes.

#### MANUAL TESTING
See the output in the Checks section for testing results.  If you so desire, you can manually launch one from the [Actions](https://github.com/broadinstitute/scp-rails-baseimage/actions/workflows/test-base-docker-image.yml) interface and select this branch as the target (`jb-node-20-upgrade`).
